tests/integration-tests-topologies/src/main/java/org/apache/pulsar/tests/topologies/PulsarCluster.java

```diff
@@ -112,6 +108,12 @@ private PulsarCluster(PulsarClusterSpec spec) {
             .withEnv("zookeeperServers", ZKContainer.NAME)
             .withEnv("configurationStoreServers", CSContainer.NAME + ":" + CS_PORT)
             .withEnv("clusterName", clusterName);
+
+        this.s3Container = new S3Container(clusterName, S3Container.NAME)
```

Comment:
It'd be better to create a subclass of PulsarCluster, PulsarClusterWithS3, and only start S3 in that case. As it is now, S3 is being started for all cases.

Reply:
a cluster is built from a spec. an alternative is to add a flag `enableTieredStorage()` in the cluster spec. the cluster can start s3 container if tiered storage is enabled.